File: bench/interp/bench_strops.js
const line = "alpha,beta,gamma,delta,epsilon,zeta,eta,theta";
const n = 120000;
const t0 = Date.now();
let total = 0;
for (let i = 0; i < n; i++) {
    const parts = line.split(",");
    total += parts.length;
    if (line.indexOf("delta") >= 0) total += 1;
    const joined = parts.join("|");
    total += joined.length;
}
console.log("result: " + total);
console.log("elapsed: " + (Date.now() - t0) + " ms");

File: bench/interp/bench_strops.py
import time
line = "alpha,beta,gamma,delta,epsilon,zeta,eta,theta"
n = 120000
t0 = time.perf_counter()
total = 0
for _ in range(n):
    parts = line.split(",")
    total += len(parts)
    if line.find("delta") >= 0: total += 1
    joined = "|".join(parts)
    total += len(joined)
print("result: %d" % total)
print("elapsed: %d ms" % int((time.perf_counter()-t0)*1000))
